### PR TITLE
Start listening for events on child elements. (amp-analytics)

### LINKED ISSUE #1761: span selector not working in amp-analytics
From @Polylike:

> Hey, it looks like there are a other problem. The problems comes with the tricker. I'm trying this:`"link-trigger":{"on": "click", "selector": "a", "request": "event", "vars":{"eventCategory": "Link", "eventAction": "clicked-link"}},`

> Works fine and every click gets tracked correct. 

> Trying this:
`"span-trigger":{"on": "click", "selector": "#span-intern", "request": "event", "vars":{"eventCategory": "Span", "eventAction": "clicked-span"}}, `

> in combination with this

> ` <span id="span-intern">
    Click here to generate an event
  </span>`

> nothing happend. 

> Looks like there is a problem with the "selector": "#span-intern".

### PR DESCRIPTION
**Pull request by avimehta**
/cc @btownsend

Fixes #1761

### CODE REVIEW

**Comment by rudygalfi** [comment]
@dvoytenko Can you review?

File: extensions/amp-analytics/0.1/instrumentation.js [hunk 1]
```diff
@@ -207,9 +207,23 @@ class InstrumentationService {
...
         listener(new AnalyticsEvent(AnalyticsEventType.CLICK));
+      } else {
+        // More expensive search.
+        let el = e.target;
+        while (el.parentElement != null && el.parentElement.tagName != 'BODY') {
+          el = el.parentElement;
+          if (this.matchesSelector_(el, selector)) {
+            listener(new AnalyticsEvent(AnalyticsEventType.CLICK));
+            // Don't fire the event multiple times even if the more than one
+            // ancestors match the selector.
```
Comment (dvoytenko) [inline extensions/amp-analytics/0.1/instrumentation.js]: "more than one ancestor matches the selector".

File: extensions/amp-analytics/0.1/test/test-instrumentation.js [hunk 2]
```diff
@@ -106,6 +106,45 @@ describe('instrumentation', function() {
...
+    el3.appendChild(el1);
+
+    const el4 = document.createElement('div');
+    el4.className = 'x';
+    el4.id = 'y';
+    el4.appendChild(el3);
+    el4.appendChild(el2);
+
+    const fnClassX = sandbox.stub();
+    addListener(window, {'on': 'click', 'selector': '.x'}, fnClassX);
```
Comment (dvoytenko) [inline extensions/amp-analytics/0.1/test/test-instrumentation.js]: Please rebase. I redid these tests a bit to do `ins.addListener` to improve test isolation - there were a few flakes.

**Comment by dvoytenko** [comment]
Looks good. Just a couple of small nits.

File: extensions/amp-analytics/0.1/instrumentation.js (hunk 1 above)
Comment (avimehta) [inline extensions/amp-analytics/0.1/instrumentation.js]: done.

File: extensions/amp-analytics/0.1/test/test-instrumentation.js (hunk 2 above)
Comment (avimehta) [inline extensions/amp-analytics/0.1/test/test-instrumentation.js]: done.

**Comment by dvoytenko** [comment]
LGTM. Merging.

**Pull request merged**